tests/test_anneal.py

```diff
@@ -65,18 +66,18 @@ def test_auto():
     assert tsp.updates == auto_schedule['updates']
 
 
-def test_state(tmpdir):
+def test_state():
     # initial state, a randomly-ordered itinerary
     init_state = list(cities.keys())
     random.shuffle(init_state)
 
     tsp = TravellingSalesmanProblem(init_state, distance_matrix)
     tsp.copy_strategy = "slice"
 
-    statefile = str(tmpdir.join("state.pickle"))
-    tsp.save_state(fname=statefile)
+    with tempfile.NamedTemporaryFile('w') as statefile:
```

Comment:
👍  - one question though:

 is this necessary to run tests with setup.py or ? Just trying to understand why this change is important.

Reply:
Things seem to work without this when you use ```pytest```. Initially I didn't understand how the tests were supposed to be run, so I was trying to use ```nosetests```, which did not support this "magical" tmpdir argument. I can drop this change if you want.